uitools/toolkitwidgets.cmake

```diff
@@ -11,56 +11,165 @@
 # limitations under the License.
-include(${CMAKE_CURRENT_LIST_DIR}/common.cmake)
+function(setup_tookitwidgets TARGET_NAME TOOLKIT_DIR)
+    set(TOOLKITUI_PATH ${TOOLKIT_DIR}/uitools)
+    set(CPP_PATH ${TOOLKITUI_PATH}/cpp/Esri/ArcGISRuntime/Toolkit)
+    set(WIDGET_PATH ${TOOLKITUI_PATH}/widgets/Esri/ArcGISRuntime/Toolkit)
 
-set(WIDGETPATH ${CMAKE_CURRENT_LIST_DIR}/widgets/Esri/ArcGISRuntime/Toolkit)
+    setup_common_vars(TOOLKITCOMMON_HEADERS TOOLKITCOMMON_SOURCES)
+    setup_widget_vars(TOOLKITWIDGET_HEADERS TOOLKITWIDGET_SOURCES)
+    setup_forms_vars(TOOLKITWIDGET_FORMS)
 
-set(TOOLKITWIDGET_SOURCES
-    ${WIDGETPATH}/AuthenticationView.cpp
-    ${WIDGETPATH}/BasemapGallery.cpp
-    ${WIDGETPATH}/BookmarksView.cpp
-    ${WIDGETPATH}/CoordinateConversion.cpp
-    ${WIDGETPATH}/FloorFilter.cpp
-    ${WIDGETPATH}/NorthArrow.cpp
-    ${WIDGETPATH}/OverviewMap.cpp
-    ${WIDGETPATH}/Internal/ClientCertificatePasswordDialog.cpp
-    ${WIDGETPATH}/Internal/ClientCertificateView.cpp
-    ${WIDGETPATH}/Internal/CoordinateEditDelegate.cpp
-    ${WIDGETPATH}/Internal/Flash.cpp
-    ${WIDGETPATH}/Internal/OAuth2View.cpp
-    ${WIDGETPATH}/Internal/SslHandshakeView.cpp
-    ${WIDGETPATH}/Internal/UserCredentialView.cpp)
+    target_include_directories(${TARGET_NAME} PRIVATE
+        ${TOOLKITUI_PATH}/cpp
+        ${TOOLKITUI_PATH}/widgets
+        ${CPP_PATH}
+        ${WIDGET_PATH})
 
-set(TOOLKITWIDGET_RESOURCES ${CMAKE_CURRENT_LIST_DIR}/images/esri_arcgisruntime_toolkit_images.qrc)
+    target_sources(${TARGET_NAME} PRIVATE
+        ${TOOLKITCOMMON_HEADERS}
+        ${TOOLKITCOMMON_SOURCES}
+        ${TOOLKITWIDGET_HEADERS}
+        ${TOOLKITWIDGET_SOURCES}
+        ${TOOLKITWIDGET_FORMS}
+        ${TOOLKITUI_PATH}/images/esri_arcgisruntime_toolkit_images.qrc)
 
-set(TOOLKITWIDGET_FORMS
-    ${WIDGETPATH}/AuthenticationView.ui
-    ${WIDGETPATH}/BasemapGallery.ui
-    ${WIDGETPATH}/BookmarksView.ui
-    ${WIDGETPATH}/CoordinateConversion.ui
-    ${WIDGETPATH}/FloorFilter.ui
-    ${WIDGETPATH}/OverviewMap.ui
-    ${WIDGETPATH}/Internal/ClientCertificatePasswordDialog.ui
-    ${WIDGETPATH}/Internal/ClientCertificateView.ui
-    ${WIDGETPATH}/Internal/OAuth2View.ui
-    ${WIDGETPATH}/Internal/SslHandshakeView.ui
-    ${WIDGETPATH}/Internal/UserCredentialView.ui)
+    find_package(Qt6 COMPONENTS REQUIRED Widgets WebEngineWidgets Svg)
 
-target_include_directories(${PROJECT_NAME} PRIVATE 
-    ${CMAKE_CURRENT_LIST_DIR}/widgets
-    ${WIDGETPATH})
+    target_link_libraries(${TARGET_NAME} PRIVATE
+        Qt::Widgets
+        Qt::WebEngineWidgets
+        Qt::Svg)
 
-target_sources(${PROJECT_NAME} PRIVATE
-    ${TOOLKITCOMMON_SOURCES}
-    ${TOOLKITWIDGET_SOURCES}
-    ${TOOLKITWIDGET_RESOURCES}
-    ${TOOLKITWIDGET_FORMS})
+    target_compile_definitions(${TARGET_NAME} PRIVATE WIDGETS_ARCGISRUNTIME_TOOLKIT)
+endfunction()
 
-find_package(Qt6 COMPONENTS REQUIRED Widgets WebEngineWidgets Svg)
+function(setup_common_vars HEADERS SOURCES)
```

Comment:
I know this is repeated in both `toolkitwidgets.cmake` and `toolkitcpp.cmake`, I wanted to put this in `common.cmake`, but then the `setup_common_vars` function would either need to be passed the CPP_PATH as an argument, or rely on CPP_PATH being defined by the times its called in its parent scope.

Reply:
Ok I see what you are saying. I think it is simplest to just list the files necessary, no reason to overcomplicate.